atlp-rwanda/e-comm-team-techsmith-fn
# #185548712 Add DM features to chatroom

## What does this PR do?

This PR adds a Direct Messages feature to the application's chatroom.

## Description of Task to be completed?

- Two users can chat privately using Direct Messages
- User can join different groups
- Messages shared to a group can only be seen by respective group members
- User can see whether another party is active when chatting.
- User can see a list of active users in a group.

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout ft-chatroom-dm-185548712`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## What are the relevant pivotal tracker/Trello stories?

- [#185548712](https://www.pivotaltracker.com/story/show/185548712)

## Screenshots (if appropriate)
<img width="1440" alt="Screenshot 2023-07-16 at 18 54 30" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/ee210742-d366-4f8c-9d9e-4889f55089e4">
<img width="1440" alt="Screenshot 2023-07-18 at 01 12 50" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/44ad2a1f-c571-446a-ab48-dd31b28ab1ea">
<img width="1440" alt="Screenshot 2023-07-18 at 01 13 19" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/e857d1f0-c235-4843-99ea-bfde59bdbdbe">
<img width="1440" alt="Screenshot 2023-07-18 at 01 13 37" src="https://github.com/atlp-rwanda/e-comm-team-techsmith-fn/assets/31816352/111256ed-b640-40af-9093-06c033cc7824">
